### PR TITLE
Python3 converting

### PR DESCRIPTION
**Pull request by paledega**
LMDE installer python3 convert process.

Current status:
-Gui works without bug
-Some installation process are broken

Python2 end of the life and slow. We need python3 version of installer.

### CODE REVIEW

File: usr/lib/live-installer/frontend/gtk_interface.py [hunk 1]
```diff
@@ -641,18 +641,17 @@ def assign_language(self, treeview, data=None):
         ''' Called whenever someone updates the language '''
         model = treeview.get_model()
         selection = treeview.get_selection()
-        if selection.count_selected_rows > 0:
```
Comment (paledega) [inline usr/lib/live-installer/frontend/gtk_interface.py:644]: It not related with python3. I think

File: usr/lib/live-installer/installer.py [hunk 2]
```diff
@@ -20,12 +20,12 @@ def __init__(self, setup):
         self.setup = setup
 
         # Flush print when it's called
-        sys.stdout = os.fdopen(sys.stdout.fileno(), 'w', 0)
+        #sys.stdout = os.fdopen(sys.stdout.fileno(), 'w', 0)
```
Comment (paledega) [inline usr/lib/live-installer/installer.py:23]: It is not needed for python3

**Comment by TheDoop** [comment]
Nice

**Comment by paledega** [comment]
We fixed gui bugs
now we must fix installation process bugs

**Comment by paledega** [comment]
image

I am testing on LMDE4 iso.
We have only byte-like object and str issue

**Comment by sulincix** [comment]
Ohh
@paledega good job

File: usr/lib/live-installer/partitioning.py [hunk 3]
```diff
@@ -277,7 +278,7 @@ def __init__(self):
             lvm_partitions = disk.getLVMPartitions()
             print("           -> %d LVM partitions" % len(lvm_partitions))
 
-            partition_set = set(free_space_partition + primary_partitions + logical_partitions + raid_partitions + lvm_partitions)
+            partition_set = tuple(free_space_partition + primary_partitions + logical_partitions + raid_partitions + lvm_partitions)
```
Comment (paledega) [inline usr/lib/live-installer/partitioning.py:281]: this changes fixed type error bug

File: usr/lib/live-installer/partitioning.py [hunk 4]
```diff
@@ -453,7 +454,7 @@ def __init__(self, partition):
         try:
             print("                  . About to mount it...")
             os.system('mount --read-only {} {}'.format(self.path, TMP_MOUNTPOINT))
-            size, free, self.used_percent, mount_point = getoutput("df {0} | grep '^{0}' | awk '{{print $2,$4,$5,$6}}' | tail -1".format(self.path)).split(None, 3)
+            size, free, self.used_percent, mount_point = str(getoutput("df {0} | grep '^{0}' | awk '{{print $2,$4,$5,$6}}' | tail -1".format(self.path)).split(None, 3))
```
Comment (paledega) [inline usr/lib/live-installer/partitioning.py:457]: moun_point variable must be string. not byte-like object

File: usr/lib/live-installer/partitioning.py [hunk 5]
```diff
@@ -468,11 +469,11 @@ def __init__(self, partition):
             print("                  . About to find more about it...")
             self.size = to_human_readable(int(size)*1024)  # for mountable partitions, more accurate than the getLength size above
             self.free_space = to_human_readable(int(free)*1024)  # df returns values in 1024B-blocks by default
-            self.used_percent = self.used_percent.strip('%') or 0
+            self.used_percent = self.used_percent.strip(b'%') or 0
```
Comment (paledega) [inline usr/lib/live-installer/partitioning.py:472]: split value must be a byte-like character

File: usr/lib/live-installer/partitioning.py [hunk 6]
```diff
@@ -76,7 +76,7 @@ def get_disks():
     lsblk = shell_exec('LC_ALL=en_US.UTF-8 lsblk -rindo TYPE,NAME,RM,SIZE,MODEL | sort -k3,2')
     for line in lsblk.stdout:
         try:
-            elements = str(line).strip().split(" ", 4)
+            elements = str(line).strip().split(" ")
```
Comment (paledega) [inline usr/lib/live-installer/partitioning.py:79]: this is better than old code I think

**Comment by paledega** [comment]
I removed webkit based parts and keyboard preview.

**Comment by paledega** [comment]
main installation empty now. We should add image.

**Comment by paledega** [comment]
https://github.com/linuxmint/live-installer/pull/114/commits/62fcfc83ec47db0d22b1ac4f7918050b2f0e5c15
this commit bugless and full feature code. newest commits are remove some parts. I am gonna add again.

**Comment by sulincix** [comment]
We are forking live-intaller called as "17g"

image

I think Linux mint (ubuntu or debian edition) can use our fork.

source code: https://gitlab.com/ggggggggggggggggg/17g

**Comment by sulincix** [comment]
https://github.com/linuxmint/live-installer/issues/117

**Review by JulianGro — CHANGES_REQUESTED** [review]
I was wondering why this hasn't been merged yet.
After a quick look:
    - debian/ folder has been removed
    - the installation slideshow has been removed
    - keyboard layout preview images have been removed?
    - systemd services that used by the live system have been removed
    - generally a lot of changes that are unrelated to a Python3 upgrade and should really be in different PRs
Now I know why this hasn't been merged yet..

**Comment by clefebvre** [comment]
It's hard to merge PRs when they're not modular.

This will help though. Thanks @paledega, @sulincix and everyone involved.

**Pull request closed without merging**